src/components/RandomBeer.js

```diff
@@ -0,0 +1,40 @@
+import React, { useState } from 'react';
+import Header from './Header';
+import axios from 'axios';
+import OneBeer from './OneBeer';
+
+function RandomBeer() {
+  const [randomBeer, setRandomBeer] = useState({});
+  const [loaded, setLoaded] = useState(false);
+
+  function getRandomBeer() {
+    axios
+      .get(`https://ih-beers-api2.herokuapp.com/beers/random`)
+      .then((beerData) => {
+        setRandomBeer(beerData.data);
+      })
+      .catch((error) => console.error(`Something went wrong: ${error}`));
+  }
+
+  if (loaded) {
+    return (
+      <div>
+        <Header />
+        <div className="beerWrapper">
+          <OneBeer beer={randomBeer} />
+        </div>
+      </div>
+    );
+  } else {
+    getRandomBeer();
+    setLoaded(true);
+    return (
+      <div>
+        <Header />
+        <div>Loading beer</div>
+      </div>
+    );
+  }
```

Comment:
Good work, 

I replaced line 19-36 with the following code (using `useEffect` so the conditional rendering is easier to type). This is to give you an alternative perspective, you don't have to implement it.
```
useEffect(()=>{
    if(!loaded) {
      getRandomBeer();
      setLoaded(true);
    }
  }, [loaded])

  return (
    <div>
      <Header />
      <div className="beerWrapper">
       {loaded ? <OneBeer beer={randomBeer} /> : <div>Loading beer</div>} 
      </div>
    </div>
  );

Reply:
Thanks Lloyd! I was struggling with getting useState to work so it's good to have an example that does.